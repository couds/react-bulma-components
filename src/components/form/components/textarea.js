import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


import CONSTANTS from '../../../constants';
import events from '../../../modifiers/events';

const colors = [null].concat(Object.keys(CONSTANTS.COLORS).map(key => CONSTANTS.COLORS[key]));

export default class Textarea extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    style: PropTypes.object,
    /**
     * The size of button using the classes from Bulma
     */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    /**
     * The color of button using the classes from Bulma
     */
    color: PropTypes.oneOf(colors),
    /**
     * True if this element is read-only
     */
    readOnly: PropTypes.bool,
    /**
     * True disable this element
     */
    disabled: PropTypes.bool,
    /**
     * The text to be displayed if the input is empty
     */
    placeholder: PropTypes.string,
    /**
     * The value of the input field
     */
    value: PropTypes.string,
    /**
     * Number of rows of this textarea
     */
    rows: PropTypes.number,
    /**
     * The name of the input field Commonly used for [multi-input handling](https://reactjs.org/docs/forms.html#handling-multiple-inputs)
     */
    name: PropTypes.string,
    ...events.propTypes,
  }

  static defaultProps = {
    className: '',
    style: {},
    size: null,
    color: null,
    readOnly: false,
    disabled: false,
    placeholder: '',
    rows: 4,
    value: '',
    name: '',
    ...events.defaultProps,
  }

  render() {
    const {
      className,
      style,
      size,
      color,
      readOnly,
      disabled,
      placeholder,
      rows,
      value,
      name,
      ...props
    } = this.props;
    return (
      <textarea
        name={name}
        {...events.props(props)}
        value={value}
        rows={rows}
        style={style}
        placeholder={placeholder}
        readOnly={readOnly}
        disabled={disabled}
        className={classnames('textarea', className, {
        [`is-${size}`]: size,
        [`is-${color}`]: color,
      })}
      />
    );
  }
}
