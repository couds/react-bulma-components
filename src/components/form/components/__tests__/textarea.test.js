import React from 'react';
import { render } from 'react-testing-library';
import { Textarea } from '../textarea';

describe('Textarea component', () => {
  it('should exist', () => {
    expect(Textarea).toMatchSnapshot();
  });
  it('should have textarea classname', () => {
    const { asFragment } = render(<Textarea />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with Bulma classname', () => {
    const { asFragment } = render(<Textarea className="other-class this-is-a-test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(<Textarea style={{ height: 250 }} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be large with 10 rows and readOnly', () => {
    const { asFragment } = render(<Textarea rows={10} readOnly size="large" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be disabled with placeholder and value', () => {
    const { asFragment } = render(<Textarea value="TEST" disabled placeholder="hello tests" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
