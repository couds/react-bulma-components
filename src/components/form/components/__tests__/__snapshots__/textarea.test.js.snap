// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component should be disabled with placeholder and value 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
    disabled=""
    name=""
    placeholder="hello tests"
    rows="4"
  >
    TEST
  </textarea>
</DocumentFragment>
`;

exports[`Textarea component should be large with 10 rows and readOnly 1`] = `
<DocumentFragment>
  <textarea
    class="textarea is-large"
    name=""
    placeholder=""
    readonly=""
    rows="10"
  />
</DocumentFragment>
`;

exports[`Textarea component should concat classname in props with Bulma classname 1`] = `
<DocumentFragment>
  <textarea
    class="textarea other-class this-is-a-test"
    name=""
    placeholder=""
    rows="4"
  />
</DocumentFragment>
`;

exports[`Textarea component should exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "backgroundColor": null,
    "className": "",
    "clearfix": false,
    "clipped": false,
    "color": null,
    "disabled": false,
    "hidden": false,
    "invisible": false,
    "italic": false,
    "marginless": false,
    "name": "",
    "overlay": false,
    "paddingless": false,
    "placeholder": "",
    "pull": null,
    "radiusless": false,
    "readOnly": false,
    "responsive": undefined,
    "rows": 4,
    "shadowless": false,
    "size": null,
    "style": Object {},
    "textAlignment": null,
    "textColor": null,
    "textSize": null,
    "textTransform": null,
    "textWeight": null,
    "unselectable": false,
    "value": "",
  },
  "propTypes": Object {
    "backgroundColor": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "color": [Function],
    "disabled": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "name": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "placeholder": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "readOnly": [Function],
    "responsive": [Function],
    "rows": [Function],
    "shadowless": [Function],
    "size": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
    "value": [Function],
  },
  "render": [Function],
}
`;

exports[`Textarea component should have textarea classname 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
    name=""
    placeholder=""
    rows="4"
  />
</DocumentFragment>
`;

exports[`Textarea component should use inline styles 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
    name=""
    placeholder=""
    rows="4"
    style="height: 250px;"
  />
</DocumentFragment>
`;
